packages/logging-winston/src/index.js

```diff
@@ -126,6 +131,30 @@ winston.transports.StackdriverLogging = LoggingWinston;
 util.inherits(LoggingWinston, winston.Transport);
 
 /**
+ * Gets the current fully qualified trace ID when available from the
+ * @google-cloud/trace-agent library in the LogEntry.trace field format of:
+ * "projects/[PROJECT-ID]/traces/[TRACE-ID]".
+ */
+function getCurrentTraceFromAgent() {
```

Comment:
@stephenplusplus Instead of duplication between `logging-{winston,bunyan}`, it would be nice to have this in `logging`. WDYT?

Reply:
That's okay with me.

Reply:
Cool. I just added a pull request for the `logging` library for that: https://github.com/GoogleCloudPlatform/google-cloud-node/pull/2597

I think we will need to release a new `logging` npm module version and then update `logging-winston` and `logging-bunyan` to depend on it as part of this change.

Reply:
A second thought on this (I'm sorry) is maybe this code belongs in a different layer. `@google-cloud/logging` is a consumer module, as opposed to a utility suite. Bending the rules seems *okay*, although I could see an accident where a future maintainer removes this method because it's not clear why it's there (since the `@google-cloud/logging` codebase never directly calls it). And similarly, if the logging-* modules decide not to use this method anymore, its removal from `@google-cloud/logging` could be forgotten.

Alternatively, I've thought of a couple solutions. More are welcome:

- (A) Put this code in `@google-cloud/common`, i.e. `require('@google-cloud/common').util.getCurrentTraceFromAgent()`.
  - 👍 The code is in a utility library
  - 👎 The code isn't useful for all API modules, just `logging-bunyan` and `logging-winston`
- (B) Duplicate the code, as it is in this PR.
  - 👍 Code relevant to `logging-bunyan` and `logging-winston` stays close to their respective codebases.
  - 👎 Code is duplicated, which could mean a patch in one is forgotten to be placed in the other.
- (C) ?

I'm leaning towards (B), as the duplication is minor. And as is evident by this PR, usually a modification to `logging-winston` means a modification to `logging-bunyan` is simultaneous.

Reply:
I think for (C), the eventual best solution would probably be to separate out a context propagation library that the trace SDK, logging-{winston,bunyan} can all three depend on that would expose this "get current context id" functionality. But that's a longer term approach not something that will help us directly here in this PR.

I have heard it said that "duplication is better than the wrong abstraction", and the actual function we would be duplicating is pretty small and straightforward.

Reply:
@ofrobots, are you OK with keeping the getCurrentTraceFromAgent duplicated in both logging-winston and logging-bunyan for now?